lib/templates/lab2/lab2.dart

```diff
@@ -0,0 +1,81 @@
+class Book with JsonSerializable {
+  final String title;
+  final String author;
+  int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+
+  void bookDetails() {
+    print(
+        'The name of the book is $title. The author is $author. The number of pages is $numberOfPages.');
+  }
+}
+
+class FictionBook extends Book {
+  String genre;
+  FictionBook(super.title, super.author, super.numberOfPages, this.genre);
+  @override
+  void bookDetails() {
+    print(
+        'The name of the book is $title. The author is $author. The number of pages is $numberOfPages. The genre is $genre.');
+  }
+}
+
+mixin JsonSerializable {
+  String? title;
+  String? author;
+  int? numberofPages;
+  Map<String, dynamic> toJson() {
+    return {'title': title, 'author': author, 'number of pages': numberofPages};
+  }
+}
+
+double calculateArea(double length, double width,
+    [String shape = 'rectangle']) {
+  if (shape == 'rectangle') {
+    return length * width;
+  }
+  if (shape == 'circle') {
+    return 3.14 * length * length;
+  }
+  return (length * width) / 2; // triangle
+}
+
+class User {
+  String name;
+  int? age;
+  String? email;
```

Comment:
These parameters are positional, so you cannot use user without them in constructor, so I can't create user without email 
Name has to be required